src/program/snabbnfv/doc/getting-started.md

```diff
@@ -0,0 +1,610 @@
+```
+ 
+If all went well so far, you can finally ping between both VM's. If you used non-Linux virtual machines for this test, e.g. [OpenBSD](http://www.openbsd.org), you might not be able to send or receive packets within the guest OS. This issue can be solved (for OpenBSD 5.7 at least) by forcing qemu to use vhost (vhostforce=on):
+
+```
+$ sudo /usr/local/bin/qemu-system-x86_64 \
+  -drive if=virtio,file=/home/mwiget/openbsd1.qcow2 -M pc -smp 1 \
+  --enable-kvm -cpu host -m 1024 -numa node,memdev=mem \
+  -object memory-backend-file,id=mem,size=1024M,mem-path=/mnt/huge,share=on \
+  -chardev socket,id=char0,path=/home/mwiget/vhost-sockets/vm1.socket,server \
+  -netdev type=vhost-user,id=net0,chardev=char0,vhostforce=on  \
+  -device virtio-net-pci,netdev=net0,mac=52:54:00:00:00:01 \
+  -vnc :1
+```
+
+The snabbnfv terminals will show counter output similar to:
+
+```
+link report:
+                  14 sent on id1_NIC.tx -> id1_Virtio.rx (loss rate: 0%)
+                  14 sent on id1_Virtio.tx -> id1_NIC.rx (loss rate: 0%)
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+```
+
+and
+
+```
+link report:
+                 100 sent on id2_NIC.tx -> id2_Virtio.rx (loss rate: 0%)
+                 100 sent on id2_Virtio.tx -> id2_NIC.rx (loss rate: 0%)
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 4         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+link report:
+                 120 sent on id2_NIC.tx -> id2_Virtio.rx (loss rate: 0%)
+                 120 sent on id2_Virtio.tx -> id2_NIC.rx (loss rate: 0%)
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+load: time: 1.00s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 98   sleep: 100 us
+```
+
+The difference in packet counters is a result of me stopping and starting one of the snabbnfv processes mid-flight. According to the documentation thats ok and it does indeed work just fine. 
```

Comment:
Little background: Yes, you can restart the Snabb Switch process and it will reconnect to the guests and continue to provide their networking. This is currently the one-and-only change on the `v2.1.0-vhostuser` branch that is not upstream in QEMU. (The other changes on that branch have landed upstream in QEMU after the 2.1 release.)

This guide should also work with the latest upstream QEMU with the exception that restarting the Snabb Switch and continuing to do networking will only work if you apply commit [f393aea](https://github.com/SnabbCo/qemu/commit/f393aea2301734647fdf470724433f44702e3fb9). That change requires some thinking about how to do "right" for QEMU, the trouble is that there is no analogous situation where you reboot the kernel underneath a running QEMU process. The other two changes on our branch are already upstream in QEMU.